kafka-client/src/commonMain/kotlin/com/icemachined/kafka/common/serialization/JsonSerializer.kt

```diff
@@ -0,0 +1,28 @@
+package com.icemachined.kafka.common.serialization
+
+import com.icemachined.kafka.clients.consumer.Headers
+import com.icemachined.kafka.common.header.KafkaHeaders
+import com.icemachined.kafka.common.header.RecordHeader
+
+import kotlin.reflect.KType
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+
+/**
+ * JsonSerializer
+ */
+class JsonSerializer<T>(
+    private val typesMap: Map<String, KType>
+) : Serializer<T> {
+    override fun serialize(
+        data: T,
+        topic: String?,
+        headers: Headers?
+    ): ByteArray? {
+        val typeName = data!!::class.qualifiedName
```

Comment:
since now `T` cannot be null, maybe change `Serializer<T>` to `Serializer<T : Any>`?

Reply:
fixed

Reply:
I don't see that changes)

Reply:
class JsonSerializer<T : Any> for inherited interface we can't do so it takes generic rules from declaration

Reply:
I don't see a need to restrict on interface level, you can always restrict on implementation level. 
Actually we need to allow null both in key and value in Serializer and Deserializer. Added check for serializer